src/services/__tests__/redisSessionStorage.test.ts

```diff
@@ -1626,3 +1632,166 @@ describe("RedisSessionStorage#setPagoPaNoticeEmail", () => {
     expect(response).toEqual(E.left(expectedError));
   });
 });
+
+describe("RedisSessionStorage#getLollipopAssertionRefForUser", () => {
+  it("should success and return an assertionRef", async () => {
+    mockGet.mockImplementationOnce((_, callback) =>
+      callback(null, anAssertionRef)
+    );
+    const response = await sessionStorage.getLollipopAssertionRefForUser(
+      aValidUser
+    );
+
+    expect(mockGet).toHaveBeenCalledTimes(1);
+    expect(mockGet.mock.calls[0][0]).toBe(`KEYS-${aValidUser.fiscal_code}`);
+    expect(E.isRight(response)).toBeTruthy();
+    if (E.isRight(response)) expect(response.right).toEqual(anAssertionRef);
+  });
+
+  it("should fail with a left response if an error occurs on redis", async () => {
+    const expectedError = new Error("redis Error");
+    mockGet.mockImplementationOnce((_, callback) => callback(expectedError));
+    const response = await sessionStorage.getLollipopAssertionRefForUser(
+      aValidUser
+    );
+
+    expect(mockGet).toHaveBeenCalledTimes(1);
+    expect(mockGet.mock.calls[0][0]).toBe(`KEYS-${aValidUser.fiscal_code}`);
+    expect(E.isLeft(response)).toBeTruthy();
+    if (E.isLeft(response)) expect(response.left).toEqual(expectedError);
+  });
+
+  it("should fail with a left response if the value stored is invalid", async () => {
+    mockGet.mockImplementationOnce((_, callback) =>
+      callback(null, "an invalid value")
+    );
+    const response = await sessionStorage.getLollipopAssertionRefForUser(
+      aValidUser
+    );
+
+    expect(mockGet).toHaveBeenCalledTimes(1);
+    expect(mockGet.mock.calls[0][0]).toBe(`KEYS-${aValidUser.fiscal_code}`);
+    expect(E.isLeft(response)).toBeTruthy();
+  });
+});
+
+describe("RedisSessionStorage#setLollipopAssertionRefForUser", () => {
+  it("should success if the response from redis is OK on set command with default ttl", async () => {
+    mockSet.mockImplementationOnce((_, __, ___, ____, callback) =>
+      callback(null, "OK")
+    );
+    const response = await sessionStorage.setLollipopAssertionRefForUser(
+      aValidUser,
+      anAssertionRef
+    );
+
+    expect(mockSet).toHaveBeenCalledTimes(1);
+    expect(mockSet.mock.calls[0][0]).toBe(`KEYS-${aValidUser.fiscal_code}`);
```

Comment:
IMHO the use of expect(mockSet).toHaveBeenCalledWith(...) is more readable.
All the test clase use the array-style access to mock calls, so maybe we can refactor it in another PR.

Reply:
Fixed in [4806789](https://github.com/pagopa/io-backend/pull/979/commits/48067895fb38ad7ac7ecfeb073d2f3ef0a2b0f8f)
Other tests refactory will be done on another PR